content/en/docs/languages/dotnet/metrics/best-practices.md

```diff
@@ -0,0 +1,451 @@
+[Delta Aggregation Temporality](/docs/specs/otel/metrics/data-model/#temporality)
+is used, it is possible to choose a smaller cardinality limit because the SDK
+will reclaim unused metric points.
+
+> {{% alert title="Note" %}} In SDK versions `1.7.0` - `1.9.0`, metric point
+> reclaim was an experimental feature that could be enabled by setting the
+> environment variable
+> `OTEL_DOTNET_EXPERIMENTAL_METRICS_RECLAIM_UNUSED_METRIC_POINTS=true`.
+> {{% /alert %}}
+
+### Memory preallocation
+
+OpenTelemetry .NET SDK aims to avoid memory allocation on the hot code path.
+When this is combined with [proper use of Metrics API](#metrics-api), heap
+allocation can be avoided on the hot code path.
+
+:heavy_check_mark: You should measure memory allocation on hot code path, and
+ideally avoid any heap allocation while using the metrics API and SDK,
+especially when you use metrics to measure the performance of your application
+(for example, you do not want to spend 2 seconds doing
+[garbage collection](https://learn.microsoft.com/dotnet/standard/garbage-collection/)
+while measuring an operation which normally takes 10 milliseconds).
+
+## Metrics correlation
+
+In OpenTelemetry, metrics can be correlated to
+[traces](/docs/languages/dotnet/traces/) via
+[exemplars](/docs/specs/otel/metrics/sdk/#exemplar). Check the
+[Exemplars](/docs/languages/dotnet/metrics/exemplars/) tutorial to learn more.
+
+## Metrics enrichment
+
+When metrics are being collected, they normally get stored in a
+[time series database](https://en.wikipedia.org/wiki/Time_series_database). From
+storage and consumption perspective, metrics can be multi-dimensional. Taking
+the [fruit example](#example), there are two dimensions - "name" and "color".
+For basic scenarios, all the dimensions can be reported during the
+[Metrics API](#metrics-api) invocation, however, for less trivial scenarios, the
+dimensions can come from different sources:
+
+- [Measurements](/docs/specs/otel/metrics/api/#measurement) reported via the
+  [Metrics API](#metrics-api).
+- Additional tags provided at instrument creation time. For example, the
+  [`Meter.CreateCounter<T>(name, unit, description, tags)`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.meter.createcounter)
+  overload.
+- Additional tags provided at meter creation time. For example, the
+  [`Meter(name, version, tags, scope)`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.meter.-ctor)
+  overload.
+- [Resources](/docs/specs/otel/resource/sdk) configured at the `MeterProvider`
+  level.
+- Additional attributes provided by the exporter or collector. For example,
+  [jobs and instances](https://prometheus.io/docs/concepts/jobs_instances/) in
+  Prometheus.
+
+> {{% alert title="Note" %}} Instrument level tags support is not yet
+> implemented in OpenTelemetry .NET since the
+> [OpenTelemetry Specification](/docs/specs/otel/metrics/api/#instrument) does
+> not support it. {{% /alert %}}
```

Comment:
```suggestion
{{% alert title="Note" %}} Instrument level tags support is not yet
implemented in OpenTelemetry .NET since the
[OpenTelemetry Specification](/docs/specs/otel/metrics/api/#instrument) does
not support it. {{% /alert %}}
```